src/dbnode/storage/namespace.go

```diff
@@ -592,7 +571,7 @@ func (n *dbNamespace) Write(
 		return ts.Series{}, false, err
 	}
 	series, wasWritten, err := shard.Write(ctx, id, timestamp,
-		value, unit, annotation, series.WriteOptions{})
+		value, unit, annotation)
```

Comment:
Hm wait, why are we removing WriteOptions?

By the way, just so you know WriteOptions is now going to be used by other code that needs to propagate things to the series/buffer:
https://github.com/m3db/m3/pull/1596

Reply:
ok, I will add it back. I thought this is only for schema passing.

For schema, since WriteOpts can only be used for write, not read, while nsCtx can be used for both, I have decided not to use WriteOpts to propagate down (granted, I could have something called ReadOpts, but the code path for read is more complicated than write).

It appears using series.Options to pass SchemaRegistry to series/buffer/bucket level at Reset time  is a lot less code and equally effective/explicit.